packages/mantine/src/components/modal-wizard/ModalWizard.tsx

```diff
@@ -224,8 +233,12 @@ export const ModalWizard: ModalWizardType = ({
                             if (isLastStep) {
                                 onFinish?.() ?? handleClose(false);
                             } else {
-                                onNext?.(currentStepIndex + 1);
-                                setCurrentStepIndex(currentStepIndex + 1);
+                                if (isStepValidatedOnNext && onNext) {
+                                    onNext?.(currentStepIndex, setCurrentStepIndex);
+                                } else {
+                                    onNext?.(currentStepIndex + 1);
```

Comment:
Why do we have different values for `currentStepIndex`?

Reply:
I will modify this.